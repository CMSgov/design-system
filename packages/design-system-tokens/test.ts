import { flatten } from './src/lib/utility';

const someObject = {
  colors: {
    red: 'rose',
    blue: 'cerulean',
    transparent: {
      black: '#00000088',
      white: '#FFFFFF88',
    },
  },
  heights: {
    tall: '4vh',
    short: '0.5vh',
  },
};

const result = flatten(someObject, '--');
console.log(result);
