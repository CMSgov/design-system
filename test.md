This is a random file to test the build-a-branch pipeline

more test