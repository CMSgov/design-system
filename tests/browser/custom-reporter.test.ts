import fs from 'fs';
import path from 'path';
import MyReporter from './custom-reporter';
import { Suite, TestCase, TestResult, FullResult, FullConfig } from '@playwright/test/reporter';

test('JSON report matches snapshot', () => {
  const mockConfig: FullConfig = {
    forbidOnly: false,
    fullyParallel: false,
    globalSetup: null,
    globalTeardown: null,
    maxFailures: 0,
    projects: [],
    reporter: [],
    reportSlowTests: null,
    rootDir: './',
    quiet: false,
    shard: null,
    updateSnapshots: 'missing',
    version: '1.0',
    workers: 1,
    globalTimeout: 0,
    grep: /.*/,
    grepInvert: null,
    metadata: {},
    preserveOutput: 'always',
    webServer: null,
  };

  const mockSuite = {
    title: 'Sample Suite',
    allTests: () => [
      {
        title: 'Test 1',
        location: { file: 'sample-suite.test.ts' },
        parent: { title: 'Sample Suite', parent: null },
      },
      {
        title: 'Test 2',
        location: { file: 'sample-suite.test.ts' },
        parent: { title: 'Sample Suite', parent: null },
      },
    ],
  } as unknown as Suite;

  const reporter = new MyReporter();
  reporter.onBegin(mockConfig, mockSuite);

  reporter.onTestEnd(
    { title: 'Test 1', parent: mockSuite } as TestCase,
    { status: 'passed' } as TestResult
  );
  reporter.onTestEnd(
    { title: 'Test 2', parent: mockSuite } as TestCase,
    { status: 'failed' } as TestResult
  );

  const mockResult: FullResult = {
    status: 'failed',
    startTime: new Date(),
    duration: 0,
  };
  reporter.onEnd(mockResult);

  const reportDirectory = path.resolve(__dirname, 'test-results/testing');
  const reportPath = path.resolve(reportDirectory, 'test-report.json');
  const generatedReport = JSON.parse(fs.readFileSync(reportPath, 'utf-8'));

  expect(generatedReport).toMatchSnapshot();
});
