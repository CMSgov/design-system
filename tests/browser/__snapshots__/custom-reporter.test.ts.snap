// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSON report matches snapshot 1`] = `
Object {
  "failingTests": Array [
    Object {
      "name": "Test 2",
      "path": "[Sample Suite]",
    },
  ],
  "skippedTests": Array [],
  "status": "failed",
  "summary": Object {
    "failed": 1,
    "passed": 1,
    "skipped": 0,
  },
}
`;
